addon/Simulator.jsm

```diff
@@ -0,0 +1,142 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+"use strict";
+
+const {classes:Cc, interfaces: Ci, utils: Cu} = Components;
+
+const EXPORTED_SYMBOLS = ["Simulator"];
+
+Cu.import("resource://gre/modules/FileUtils.jsm");
+Cu.import("resource://gre/modules/Log.jsm");
+
+const TELEMETRY_RAPPOR_PATH = `chrome://shield-study-rappor/content/TelemetryRappor.jsm`;
+const { TelemetryRappor } = Cu.import(TELEMETRY_RAPPOR_PATH, {});
+
+const log = createLog("Simulator", "Info");
+
+/**
+ * Create the logger.
+ * @param {string} name - Name to show in the logs.
+ * @param {string} level - Log level.
+ */
+function createLog(name, level) {
+  var logger = Log.repository.getLogger(name);
+  logger.level = Log.Level[level] || Log.Level.Debug;
+  logger.addAppender(new Log.ConsoleAppender(new Log.BasicFormatter()));
+  return logger;
+}
+
+/**
+ * Runs the simulation and writes the data into case_reports.csv.
+ * For running the simulation, the true values from case_true_values.csv
+ * are read, and RAPPOR is executed.
+ * The expected format of case_true_values.csv is {client, cohort, value}.
```

Comment:
nit: "The case_true_value.csv is expected to have multiple lines with the following format ...."

Reply:
Done